### PR TITLE
ST: Certificates & renaming class names

### PR DESCRIPTION
**Pull request by serrss**
### Type of change
- New system test
- Refactoring

### Description
System test checks certificates for Zookeeper with port `2181` and Kafka with port `9093` by using tool `openssl s_client`.
_TBH_ I also tried to create system tests with the generetion of `clients-ca.crt` ,`clients-ca.key` and `.pem` that can be used for tool `openssl s_client -connect localhost:9093 -state` to write and read messages but these actions require installing `cfssl` which complicates the task. Therefore, I would like to put this in a separate task.

### Checklist
- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by serrss** [comment]
Hm, I also need rename .yaml in resources

**Comment by serrss** [comment]
@tombentley if you have a time plz look at this PR

File: systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            Assert.assertThat(zookeeperCertificateInfo, containsString(zookeeperCertificateChain));
+            Assert.assertThat(zookeeperCertificateInfo, containsString(zookeeperServerCertificate));
+            Assert.assertThat(zookeeperCertificateInfo, containsString(protocol));
+            Assert.assertThat(zookeeperCertificateInfo, containsString(timeout));
+        });
+    }
+
+    private String getKafkaCertificate(int podIndex) {
+        String kafkaPodName = kafkaPodName(CLUSTER_NAME, podIndex);
+        return kubeClient.execInPod(kafkaPodName, "/bin/bash", "-c", "openssl s_client -connect localhost:9093 -showcerts").out();
```
[the remark this answers is not in the document]
Comment (ppatierno) [reply systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java]: I would also use this option : 
`-CAfile /opt/kafka/broker-certs/cluster-ca.crt`
and then check this output : 
```
Verify return code: 0 (ok)
```

**Comment by ppatierno** [comment]
@serrss I added a comment in order to add an option to the `openssl` command for verifying the certificates chain passing the `cluster-ca.crt`. It works fine for Kafka because the certs are mounted in the `kafka` container (`/opt/kafka/broker-certs` folder).
It cannot work for Zookeeper because all certs are mounted in the `tls-sidecar` container. It means that in that case the `openssl` command needs to be executed in that container so we need an overload for `KubeClient.execInPod` method for specifying the container where running the command (currently the commands are working because they use the main one, so `kafka` or `zookeeper`).
In any case even with that, it seems that the image with `stunnel` has openssl as library (used by `stunnel`) but not as executable so we cannot running any openssl command for testing on the `tls-sidecar` container. Is it worth to add full openssl installation in the stunnel Docker image just for testing purposes ? What do you think @tombentley @scholzj

**Comment by tombentley** [comment]
Imho it's not worth adding to the image something needed only for testing. What you could do (though its ugly) to install the executable image you need via `exec("yum install -y openssl")` prior to using it. See, I said it was ugly.

**Comment by ppatierno** [comment]
Agree on not adding it to the image because just for testing, so your solution even if it's "ugly" could work. I think that running a test verifying the certificate chain on the `tls-sidecar` container is worth doing.

File: systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    private static final Logger LOGGER = LogManager.getLogger(SecurityST.class);
+    public static final String NAMESPACE = "security-cluster-test";
+
+    @Test
+    @JUnitGroup(name = "regression")
+    @KafkaFromClasspathYaml()
+    public void testCertificates() {
+        LOGGER.info("Running testCertificates {}", CLUSTER_NAME);
+
+        String kafkaCertificateChain = "s:/O=io.strimzi/CN=my-cluster-kafka\n" +
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java]: If we are already testing this, we should also test the new SAN values added by @ppatierno in one of the other PRs.

Comment (ppatierno) [reply systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java]: Yeah the #663 is now merged.

**Comment by ppatierno** [comment]
@tombentley the problem with installing openssl is that the container isn't running as root so we cannot install it.

**Comment by ppatierno** [comment]
from my tests we have the following : 

* test on port 9091 : it just works. We can run the `openssl` client on a `kafka` container passing the cluster CA and using, for client authentication, the same cert and key of the Kafka broker itself.
* test on port 9093 : we need a client cert signed with Clients CA. My idea was to generate this client cert on the `kafka` container directly but we have only the Clients cert mounted there and not the related key, so a CSR cannot be processed locally on the pod.
* test on ports 2888/3888 : the needed certs are mounted on the `tls-sidecar` container only where we should run the `openssl` client but we don't have `openssl' and we cannot install it because not root user. We could run it from `zookeeper` container (having the `openssl` command) but then we don't have the certs mounted in that container.

**Comment by serrss** [comment]
@scholzj @ppatierno Updated PR with following changes:
1. System test checks certificates by hostnames:
`my-cluster-kafka-bootstrap:9091`
`my-cluster-kafka-bootstrap:9093`
`my-cluster-zookeeper-client:2181`
`my-cluster-zookeeper-nodes:2888`
`my-cluster-zookeeper-nodes:3888`
2. System test checks certificates for all SSL/TLS ports by IP address. This is necessary because we need to be sure that all pods have a secure connection for ports which mentioned above.
3. For port 2888 we still have errors 
`read:errno=104`
`command terminated with exit code 104`
I have attached [logg.txt](https://github.com/strimzi/strimzi-kafka-operator/files/2245259/logg.txt) file to this PR

**Comment by serrss** [comment]
@ppatierno the same issue with `read:errno=104` if use `my-cluster-zookeeper-nodes:2888` for `openssl`

**Comment by serrss** [comment]
@ppatierno @scholzj I updated this PR by recommendations from @ppatierno 
Now PR has the following updates:
1. The test uses `-verify_hostname`
2. Test checks the following hostnames & ports:
- my-cluster-kafka-bootstrap:9093
- my-cluster-kafka-brokers:9091 (for every Kafka pod)
- my-cluster-kafka-brokers:9093 (for every Kafka pod)
- my-cluster-zookeeper-client:2181
- my-cluster-zookeeper-nodes:2181 (for every Zookeeper pod)
- my-cluster-zookeeper-nodes:2888 (for every Zookeeper pod)
- my-cluster-zookeeper-nodes:3888 (for every Zookeeper pod)

If you have a time, plz do a review again. thanks

File: systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            String commandForZookeeperPort3888 = generateOpenSSLCommandWithCerts(zookeeperPodName(CLUSTER_NAME, podId), "my-cluster-zookeeper-nodes", "3888");
+
+            String outputForZookeeperPort2181 =
+                    kubeClient.execInPodContainer(kafkaPodName(CLUSTER_NAME, podId), "kafka", "/bin/bash", "-c", commandForZookeeperPort2181).out();
+            checkZookeeperCertificates(outputForZookeeperPort2181);
+
+            try {
+                String outputForZookeeperPort2888 =
+                        kubeClient.execInPodContainer(kafkaPodName(CLUSTER_NAME, podId), "kafka", "/bin/bash", "-c", commandForZookeeperPort2888).out();
+                String outputForZookeeperPort3888 =
```
Comment (ppatierno) [inline systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java]: maybe the test against 3888 can be outside of the try/catch, we faced the problem on 2888 only right?

Comment (serrss) [reply systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java]: Oh, yes!

**Pull request merged**